core/app/[locale]/(default)/account/addresses/_actions/create-address.ts

```diff
@@ -249,11 +249,13 @@ export async function createAddress(prevState: Awaited<State>, formData: FormDat
     console.error(error);
 
     if (error instanceof BigCommerceGQLError) {
+      const formErrors = error.message.includes('Please sign in')
+        ? [t('unauthorized')]
+        : error.errors.map(({ message }) => message);
+
       return {
         ...prevState,
-        lastResult: submission.reply({
-          formErrors: error.errors.map(({ message }) => message),
-        }),
+        lastResult: submission.reply({ formErrors }),
```

Comment:
All these changes to check to see if the message includes `Please sign in` indicates to me that we probably want to return an error from the client instead, e.g. `BigCommerceAuthorizationError`. Can you explore that a bit more?

Reply:
@chanceaclark Oh definitely, good call there. Having it handled by the client would be much better. I'll explore this 👍